test/cmdlineTests/optimizer_array_sload/output

```diff
@@ -19,40 +21,61 @@ object "Arraysum_34" {
     object "Arraysum_34_deployed" {
         code {
             {
+                /// @src 0:80:429
                 mstore(64, 128)
                 if iszero(lt(calldatasize(), 4))
                 {
                     let _1 := 0
-                    if eq(0x81d73423, shr(224, calldataload(_1)))
+                    if eq(0x81d73423, /** @src 0:407:423 */ shr(224, /** @src 0:80:429 */ calldataload(_1)))
                     {
-                        if callvalue() { revert(_1, _1) }
-                        if slt(add(calldatasize(), not(3)), _1) { revert(_1, _1) }
+                        if callvalue()
+                        {
+                            /// @src 0:407:423
+                            revert(/** @src 0:80:429 */ _1, _1)
+                        }
+                        if slt(add(calldatasize(), not(3)), _1)
+                        {
+                            /// @src 0:407:423
+                            revert(/** @src 0:80:429 */ _1, _1)
+                        }
                         let var_sum := _1
-                        let var_i := _1
+                        /// @src 0:368:378
+                        let var_i := /** @src 0:80:429 */ _1
                         let _2 := sload(_1)
+                        /// @src 0:364:423
                         for { }
-                        lt(var_i, _2)
+                        /** @src 0:380:397 */ lt(var_i, _2)
+                        /// @src 0:368:378
                         {
+                            /// @src 0:407:423
                             if eq(var_i, not(0)) { panic_error_0x11() }
-                            var_i := add(var_i, 1)
+                            /** @src 0:399:402 */ var_i := /** @src 0:407:423 */ add(var_i, 1)
```

Comment:
Is this coming from the Assignment or from the Identifier on the left-hand side of the assignment? I assume the assignment itself still has `0:407:423`, otherwise it would have used `///` - can you confirm?

When we re-parse this, the parser will apply it to the assignment itself, though - I don't think we can avoid that, so the transform is not bijective, unfortunately.

Reply:
It seems my call to the sourceComment in the `Assignment` visit was too late and it visited the identifiers first and that's why we are seeing this